webpack.config.js

```diff
@@ -170,58 +119,113 @@ module.exports = [{
-    'mdc.radio': path.resolve('./packages/mdc-radio/mdc-radio.scss'),
-    'mdc.ripple': path.resolve('./packages/mdc-ripple/mdc-ripple.scss'),
-    'mdc.select': path.resolve('./packages/mdc-select/mdc-select.scss'),
-    'mdc.slider': path.resolve('./packages/mdc-slider/mdc-slider.scss'),
-    'mdc.snackbar': path.resolve('./packages/mdc-snackbar/mdc-snackbar.scss'),
-    'mdc.switch': path.resolve('./packages/mdc-switch/mdc-switch.scss'),
-    'mdc.tabs': path.resolve('./packages/mdc-tabs/mdc-tabs.scss'),
-    'mdc.textfield': path.resolve('./packages/mdc-textfield/mdc-textfield.scss'),
-    'mdc.theme': path.resolve('./packages/mdc-theme/mdc-theme.scss'),
-    'mdc.toolbar': path.resolve('./packages/mdc-toolbar/mdc-toolbar.scss'),
-    'mdc.typography': path.resolve('./packages/mdc-typography/mdc-typography.scss'),
-  },
-  output: {
-    path: OUT_PATH,
-    publicPath: PUBLIC_PATH,
-    filename: CSS_JS_FILENAME_OUTPUT_PATTERN,
-  },
-  devServer: {
-    disableHostCheck: true,
-  },
-  devtool: DEVTOOL,
-  module: {
-    rules: [{
-      test: /\.scss$/,
-      use: createCssLoaderConfig(),
-    }],
-  },
-  plugins: [
-    createCssExtractTextPlugin(),
-    createBannerPlugin(),
-  ],
 }];
 
+if (!IS_DEV) {
+  module.exports.push({
+    name: 'js-components',
+    entry: {
+      animation: [path.resolve('./packages/mdc-animation/index.js')],
+      autoInit: [path.resolve('./packages/mdc-auto-init/index.js')],
+      base: [path.resolve('./packages/mdc-base/index.js')],
+      checkbox: [path.resolve('./packages/mdc-checkbox/index.js')],
+      dialog: [path.resolve('./packages/mdc-dialog/index.js')],
+      drawer: [path.resolve('./packages/mdc-drawer/index.js')],
+      formField: [path.resolve('./packages/mdc-form-field/index.js')],
+      gridList: [path.resolve('./packages/mdc-grid-list/index.js')],
+      iconToggle: [path.resolve('./packages/mdc-icon-toggle/index.js')],
+      linearProgress: [path.resolve('./packages/mdc-linear-progress/index.js')],
+      menu: [path.resolve('./packages/mdc-menu/index.js')],
+      radio: [path.resolve('./packages/mdc-radio/index.js')],
+      ripple: [path.resolve('./packages/mdc-ripple/index.js')],
+      select: [path.resolve('./packages/mdc-select/index.js')],
+      selectionControl: [path.resolve('./packages/mdc-selection-control/index.js')],
+      slider: [path.resolve('./packages/mdc-slider/index.js')],
+      snackbar: [path.resolve('./packages/mdc-snackbar/index.js')],
+      tabs: [path.resolve('./packages/mdc-tabs/index.js')],
+      textfield: [path.resolve('./packages/mdc-textfield/index.js')],
+      toolbar: [path.resolve('./packages/mdc-toolbar/index.js')],
+    },
+    output: {
+      path: OUT_PATH,
+      publicPath: PUBLIC_PATH,
+      filename: 'mdc.[name].' + (IS_PROD ? 'min.' : '') + 'js',
+      libraryTarget: 'umd',
+      library: ['mdc', '[name]'],
+    },
+    // See https://github.com/webpack/webpack-dev-server/issues/882
```

Comment:
Is the `devServer` block here and for `css` below even necessary anymore if we're not going to run these targets when we run the dev server?

Reply:
(Additional thought, if we remove `devServer` on these, maybe move the comment that's above one of them to the first `devServer` config that's still relevant.)

Reply:
Good call! Turns out you only need one `devServer` block, and it _must_ be in the first entry of the webpack config. It is completely ignored in later entries. Removed redundant blocks.

Reply:
Seriously? Wouldn't have guessed that one... unintentionally dodged a bullet.